addons/stock/models/stock_move.py

```diff
@@ -672,15 +672,25 @@ def onchange_product_uom(self):
                 }
             }
 
-    def _search_picking_for_assignation(self):
-        self.ensure_one()
-        picking = self.env['stock.picking'].search([
+    def _search_picking_for_assignation_domain(self):
+        res = [
                 ('group_id', '=', self.group_id.id),
                 ('location_id', '=', self.location_id.id),
                 ('location_dest_id', '=', self.location_dest_id.id),
                 ('picking_type_id', '=', self.picking_type_id.id),
                 ('printed', '=', False),
-                ('state', 'in', ['draft', 'confirmed', 'waiting', 'partially_available', 'assigned'])], limit=1)
+                ('state', 'in', ['draft', 'confirmed', 'waiting',
+                    'partially_available', 'assigned'])
+                ]
+        if self.created_production_id:
```

Comment:
????

This should be in mrp. You need to override `_search_picking_for_assignation_domain` in `mrp`, but you don't need to override `_search_picking_for_assignation`